Subject: [patch] uml remove bogus __nr_sigreturn check.patch
From: Jeff Dike <jdike@addtoit.com>
Patch-mainline: 2.6.11-rc1

Before Bodo's signal fixes and my signal delivery rework, it was possible for
a process to execute UML code by running the default signal restorer.  This
is no longer possible, so this check can be removed from the sanity test for
UML accidentally tracing itself.

Signed-off-by: Jeff Dike <jdike@addtoit.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Acked-by: Gerd Knorr <kraxel@suse.de>
---
 arch/um/kernel/tt/syscall_user.c |    3 +--
 1 files changed, 1 insertion(+), 2 deletions(-)

diff -puN arch/um/kernel/tt/syscall_user.c~uml-remove-bogus-__nr_sigreturn-check arch/um/kernel/tt/syscall_user.c
--- 25/arch/um/kernel/tt/syscall_user.c~uml-remove-bogus-__nr_sigreturn-check	Fri Dec  3 13:50:45 2004
+++ 25-akpm/arch/um/kernel/tt/syscall_user.c	Fri Dec  3 13:50:45 2004
@@ -57,8 +57,7 @@ void do_syscall(void *task, int pid, int
 
 	UPT_SYSCALL_NR(TASK_REGS(task)) = PT_SYSCALL_NR(proc_regs);
 
-	if((syscall != __NR_sigreturn) &&
-	   ((unsigned long *) PT_IP(proc_regs) >= &_stext) && 
+	if(((unsigned long *) PT_IP(proc_regs) >= &_stext) &&
 	   ((unsigned long *) PT_IP(proc_regs) <= &_etext))
 		tracer_panic("I'm tracing myself and I can't get out");
 
_
